### ISSUE TITLE
Getting training and validation accuracy while training

### ISSUE THREAD

**Issue by albertqjiang** [kingoflolz/mesh-transformer-jax#128]
How could I get the training and validation per-token and per-sequence accuracy while training? Currently only the loss is logged.

Can I use the "correct" variable [here](https://github.com/kingoflolz/mesh-transformer-jax/blob/8f071d07b2c846253551eb5f0b65849902401529/mesh_transformer/layers.py#L593)? The code to create the variable looks cryptic to me: `correct = (0.0 == predicted_logits)` and I'm not sure that this generates the correct accuracy.

Thanks!!!

**Comment by kingoflolz** [comment]
The correct variable returns if the argmax is the same as the target token. This is because `predicted_logits` is calculated from `logits`, which is normalized by subtracting the maximum value (i.e. the argmax is the only element which has a zero), and only when the zero is multiplied by the "hot" element in the one-hot ground truth tensor is predicted_logits equal to zero.
